src/ai-bundle/src/AiBundle.php

```diff
@@ -462,6 +464,25 @@ private function processAgentConfig(string $name, array $config, ContainerBuilde
             $outputProcessors[] = new Reference('ai.agent.structured_output_processor');
         }
 
+        if ($config['track_token_usage']) {
+            if (!$container->hasDefinition('ai.platform.token_usage_processor.openai')) {
+                $container->setDefinition(
+                    'ai.platform.token_usage_processor.openai',
+                    new Definition(OpenAiTokenOutputProcessor::class),
+                );
+            }
+
+            if (!$container->hasDefinition('ai.platform.token_usage_processor.mistral')) {
+                $container->setDefinition(
+                    'ai.platform.token_usage_processor.mistral',
+                    new Definition(MistralTokenOutputProcessor::class),
+                );
+            }
+
+            $outputProcessors[] = new Reference('ai.platform.token_usage_processor.openai');
+            $outputProcessors[] = new Reference('ai.platform.token_usage_processor.mistral');
```

Comment:
This is wrong.

All the processors will be executed and the last one will override every others with an early return like
```
if (!\array_key_exists('usage', $content)) {
            $metadata->add('token_usage', $tokenUsage);

            return;
        }
```

We have to add the right token usage processor based on the agent model/platform.